src/main/java/seedu/address/model/animal/TaskList.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.model.animal;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the TaskList containing tasks that are addressed to the {@link Animal}
+ *
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
```

Comment:
instead of `private ArrayList<Task> taskList`, do use `private final List<Task> taskList` instead. Usage of interface as opposed to implementations as type results in looser coupling, which is more desirable.

Reply:
Got it!